### ISSUE TITLE
[PATCH] Per broker unjellyableRegistry/unjellyableFactoryRegistry

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5258]
|<img alt="titty's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| titty reported|
|-|-|
|Trac ID|trac#452|
|Type|enhancement|
|Created|2003-12-29 03:53:58Z|

Attachments:

* [patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/3af/3af0af21718ad75722b7f7dc2428e7f3f9c10432/d75b3b528276d7a9f30a04b8bccaf74e1d61f67a) (3704 bytes) - added by titty on 2003-12-29 03:53:58Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__452 452
type__enhancement enhancement
reporter__titty titty
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__wontfix wontfix
component__pb pb
keywords__ 
time__1072670038000000 1072670038000000
changetime__1470085969851774 1470085969851774
version__ 
owner__ 
cc__glyph cc__itamarst cc__warner cc__titty cc__thijs cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="titty's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|titty commented|
|-|-|

```
#!html
<pre>
This patch allows one to have a per broker
unjellyableRegistry/unjellyableFactoryRegistry instead of the global one's from
jelly.py.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
Assigning to warner
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)<a name="note_3"></a>|@warner commented|
|-|-|

the patch itself looks ok.. titty: feel like writing a test for it?

also, can you provide some details on your use case? having an example of what you might use this for will help make the docs easier to write (and thus improving the chances that it gets into trunk)

**Comment by twisted-trac** [comment]
|<img alt="titty's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|titty commented|
|-|-|

I'm using this to record the source/client of some objects, which are send via PB. It's basically the reverse use case of jellyFor(perspective,..), which btw didn't work (three years ago).

I certainly would have written some tests three years ago. Now I won't write tests unless I'm sure this will get integrated into twisted.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

Replying to [titty](#note_4):
> I'm using this to record the source/client of some objects, which are send via PB. It's basically the reverse use case of jellyFor(perspective,..), which btw didn't work (three years ago).
>
> I certainly would have written some tests three years ago. Now I won't write tests unless I'm sure this will get integrated into twisted.

It certainly won't be integrated without unit tests, looks like you already have one vote to integrate it into the trunk.

**Comment by twisted-trac** [comment]
|<img alt="titty's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|titty set status to `closed`|
|-|-|

I'm closing this as wontfix. No need to waste more time on this bug.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|GitHub commented|
|-|-|

In changeset b636ffb4afba9f1c30b91ddef95e4cef38c2dd26
```
#!CommitTicketReference repository="" revision="b636ffb4afba9f1c30b91ddef95e4cef38c2dd26"
Merge pull request #452 from twisted/8699-znerol-ascii-encoding-error-in-flatten-event

Author: znerol
Reviewer: wsanchez, rodrigc
Fixes: #8699

Fix ascii encoding error in flattenEvent()
```
